WallEView/AppDelegate.swift

```diff
@@ -0,0 +1,75 @@
+import Cocoa
+
+@NSApplicationMain
+class AppDelegate: NSObject, NSApplicationDelegate {
+
+    var eventMonitor: EventMonitor?
+
+    lazy var statusItem: NSStatusItem = {
+        let statusItem = NSStatusBar.system.statusItem(withLength: NSStatusItem.squareLength)
+        statusItem.button?.image = #imageLiteral(resourceName: "StatusBarButtonImage")
+        statusItem.button?.target = self
+        statusItem.button?.action = #selector(showContextMenu(_:))
+        statusItem.button?.sendAction(on: [.leftMouseUp, .rightMouseUp])
+        return statusItem
+    }()
+
+    lazy var statusMenu: NSMenu = {
+        let rightClickMenu = NSMenu()
+        rightClickMenu.addItem(NSMenuItem(title: "Close", action: #selector(closeApp), keyEquivalent: ""))
+        return rightClickMenu
+    }()
+
+    lazy var popover: NSPopover = NSPopover()
+
+    @objc func showContextMenu(_ sender: NSStatusBarButton) {
+        switch NSApp.currentEvent!.type {
+        case .rightMouseUp:
+            closePopover(sender: sender)
+            // even though this is deprecated, setting `statusItem.menu` results in
+            // interactions not being sent any more, not sure why
+            statusItem.popUpMenu(statusMenu)
+        case .leftMouseUp:
+            togglePopover(sender)
+        default:
+            break
+        }
+    }
+
+    @objc func togglePopover(_ sender: Any?) {
+        if popover.isShown {
+            closePopover(sender: sender)
+        } else {
+            showPopover(sender: sender)
+        }
+    }
+
+    func showPopover(sender: Any?) {
+        if let button = statusItem.button {
+            let vc: ViewController? = NSStoryboard.main.viewController()
```

Comment:
😱

Reply:
🤷‍♂ doing it in one expression with assignment does not set a generic type right (I guess extra `()` would solve it though)

Reply:
I was mostly 😱(and trolling) at the usage of IB 😜

Reply:
Couldn't use SwiftUI because I'm not on Catalina =/

Reply:
TODO: use SwiftUI